apps/events/models.py

```diff
@@ -472,18 +444,21 @@ def get_all_rules(self):
     def rule_strings(self):
         return [str(rule) for rule in self.get_all_rules()]
 
-    def satisfied(self, user, registration_start):
-
-        errors = []
+    def satisfied(
+        self, user, registration_start
+    ) -> List[Tuple[RegisterStatus, timedelta]]:
+        all_statuses: List[Tuple[RegisterStatus, timedelta]] = []
 
         for rule in self.get_all_rules():
-            response = rule.satisfied(user, registration_start)
-            if response["status"]:
-                return [response]
-            else:
-                errors.append(response)
+            status, offset = rule.satisfied(user, registration_start)
+            # Return early if any of the responses is allowed for register.
+            # We won't have to care about the rest since we know the user can register.
+            if RegisterStatus.is_allowed(status):
+                return [(status, offset)]
```

Comment:
If the user is allowed to register, is it right to assume that the offset is guaranteed to be zero, in which case we can just write 0 here?